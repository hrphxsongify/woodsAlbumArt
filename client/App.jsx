import React from 'react';
import ReactDOM from 'react-dom';
import $ from 'jquery';

class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      url: ''
    };
  }

  componentDidMount() {
    const window = document.URL;
    const url = window[window.length - 1];
    $.ajax({
      type: 'get',
      url: `ec2-34-211-225-3.us-west-2.compute.amazonaws.com/api/album/${url}`,
      success: data => {
        this.setState({ url: data[0].img });
      },
      dataType: 'json'
    });
  }

  render() {
    return (
      <div>
        <section className="playing">
          <div className="playing__art">
            <img src={this.state.url} alt="Album Art" />
          </div>

          <div className="playing__song">
            <a className="playing__song__name">Some Type of Love</a>

            <a className="playing__song__artist">Charlie Puth</a>
          </div>

          <div className="playing__add">
            <i className="ion-checkmark"></i>
          </div>
        </section>
        <section className="current-track">
          <div className="current-track__actions">
            <a className="ion-ios-skipbackward"></a>

            <a className="ion-ios-play play"></a>

            <a className="ion-ios-skipforward"></a>
          </div>

          <div className="current-track__progress">
            <div className="current-track__progress__start">0:01</div>

            <div className="current-track__progress__bar">
              <div id="song-progress"></div>
            </div>

            <div className="current-track__progress__finish">3:07</div>
          </div>
          <div className="current-track__options">
            <a href="#" className="lyrics">
              Lyrics
            </a>

            <span className="controls">
              <a href="#" className="control">
                <i className="ion-navicon"></i>
              </a>

              <a href="#" className="control">
                <i className="ion-shuffle"></i>
              </a>

              <a href="#" className="control">
                <i className="fa fa-refresh"></i>
              </a>

              <a href="#" className="control devices">
                <i className="ion-iphone"></i>
                <span>Devices Available</span>
              </a>

              <a href="#" className="control volume">
                <i className="ion-volume-high"></i>

                <div id="song-volume"></div>
              </a>
            </span>
          </div>
        </section>
      </div>
    );
  }
}

export default App;
