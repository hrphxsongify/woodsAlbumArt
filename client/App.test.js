import React from 'react';
import {
  shallow
} from 'enzyme';
import App from './App.jsx';

describe('App', () => {
  it('should render correctly', () => {
    const component = shallow( < App / > );

    expect(component).toMatchSnapshot();
  });
  it('should render two <section> tags', () => {
    const component = shallow( < App / > );
    expect(component.find("section")).toHaveLength(2);
  });
});