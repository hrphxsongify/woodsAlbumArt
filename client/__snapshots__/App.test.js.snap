// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render correctly 1`] = `
<div>
  <section
    className="playing"
  >
    <div
      className="playing__art"
    >
      <img
        alt="Album Art"
        src=""
      />
    </div>
    <div
      className="playing__song"
    >
      <a
        className="playing__song__name"
      >
        Some Type of Love
      </a>
      <a
        className="playing__song__artist"
      >
        Charlie Puth
      </a>
    </div>
    <div
      className="playing__add"
    >
      <i
        className="ion-checkmark"
      />
    </div>
  </section>
  <section
    className="current-track"
  >
    <div
      className="current-track__actions"
    >
      <a
        className="ion-ios-skipbackward"
      />
      <a
        className="ion-ios-play play"
      />
      <a
        className="ion-ios-skipforward"
      />
    </div>
    <div
      className="current-track__progress"
    >
      <div
        className="current-track__progress__start"
      >
        0:01
      </div>
      <div
        className="current-track__progress__bar"
      >
        <div
          id="song-progress"
        />
      </div>
      <div
        className="current-track__progress__finish"
      >
        3:07
      </div>
    </div>
    <div
      className="current-track__options"
    >
      <a
        className="lyrics"
        href="#"
      >
        Lyrics
      </a>
      <span
        className="controls"
      >
        <a
          className="control"
          href="#"
        >
          <i
            className="ion-navicon"
          />
        </a>
        <a
          className="control"
          href="#"
        >
          <i
            className="ion-shuffle"
          />
        </a>
        <a
          className="control"
          href="#"
        >
          <i
            className="fa fa-refresh"
          />
        </a>
        <a
          className="control devices"
          href="#"
        >
          <i
            className="ion-iphone"
          />
          <span>
            Devices Available
          </span>
        </a>
        <a
          className="control volume"
          href="#"
        >
          <i
            className="ion-volume-high"
          />
          <div
            id="song-volume"
          />
        </a>
      </span>
    </div>
  </section>
</div>
`;
